pkg/resources/dynamic_table.go

```diff
@@ -75,6 +75,34 @@ var dynamicTableShema = map[string]*schema.Schema{
 		Optional:    true,
 		Description: "Specifies a comment for the dynamic table.",
 	},
+	"refresh_mode": {
```

Comment:
Another awaiting discussion topic during resources redesign is: do we set the Snowflake default or leave it without the value? Again, both have their pros and cons. For now, for the majority of cases the default is set, so let's set it here also(for both refresh mode and initialize).